### PR TITLE
DEPR: remove Index.summary

### PR DESCRIPTION
**Pull request by jbrockmendel**
Broken off from #29725 since that has tricky mypy errors

### CODE REVIEW

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -1132,19 +1132,6 @@ def _summary(self, name=None):
...
-        Return a summarized representation.
-
-        .. deprecated:: 0.23.0
-        """
-        warnings.warn(
-            "'summary' is deprecated and will be removed in a future version.",
-            FutureWarning,
-            stacklevel=2,
-        )
-        return self._summary(name)
```
Comment (WillAyd) [inline pandas/core/indexes/base.py]: Do we need the private method any more? Not super familiar with this one but doesn't seem like something we'd use internally?

Comment (jbrockmendel) [reply pandas/core/indexes/base.py]: Yes, we use it in `DataFrame.info`

**Review by WillAyd — APPROVED** [review]
lgtm

**Comment by gfyoung** [comment]
Thanks @jbrockmendel !

**Pull request merged**